hawkbit-ddi-resource/src/main/java/org/eclipse/hawkbit/ddi/rest/resource/DdiRootController.java

```diff
@@ -266,24 +271,33 @@ private static boolean checkModule(final String fileName, final SoftwareModule m
             return ResponseEntity.notFound().build();
         }
 
+        checkAndCancelExpiredAction(action);
+
         if (!action.isCancelingOrCanceled()) {
 
             final List<DdiChunk> chunks = DataConversionHelper.createChunks(target, action, artifactUrlHandler,
                     systemManagement,
                     new ServletServerHttpRequest(requestResponseContextHolder.getHttpServletRequest()),
                     controllerManagement);
 
-            final HandlingType handlingType = action.isForce() ? HandlingType.FORCED : HandlingType.ATTEMPT;
-
             final List<String> actionHistoryMsgs = controllerManagement.getActionHistoryMessages(action.getId(),
                     actionHistoryMessageCount == null ? Integer.parseInt(DdiRestConstants.NO_ACTION_HISTORY)
                             : actionHistoryMessageCount);
 
             final DdiActionHistory actionHistory = actionHistoryMsgs.isEmpty() ? null
                     : new DdiActionHistory(action.getStatus().name(), actionHistoryMsgs);
 
+            final HandlingType downloadType = action.isForce() ? HandlingType.FORCED : HandlingType.ATTEMPT;
+            final HandlingType updateType = action.hasMaintenanceSchedule()
+                    ? (action.isMaintenanceWindowAvailable() ? downloadType : HandlingType.SKIP) : downloadType;
+
+            MaintenanceWindowStatus maintenanceWindow = action.hasMaintenanceSchedule()
+                    ? (action.isMaintenanceWindowAvailable() ? MaintenanceWindowStatus.AVAILABLE
```

Comment:
I'd use `MaintenanceWindowStatus.AVAILABLE.getStatus` to retrieve the lower case name of `available` and `unavailable` in the json-representation. Currently it's upper-case
```
{
    "id": "4",
    "deployment": {
        "download": "forced",
        "update": "forced",
        "maintenanceWindow": "AVAILABLE",
        "chunks": []
}
```